homeassistant/helpers/event.py

```diff
@@ -631,20 +633,22 @@ def _setup_entities_listener(self, domains: set[str], entities: set[str]) -> Non
         )
 
     @callback
-    def _state_added(self, event: Event) -> None:
+    def _state_added_domain(self, event: Event) -> None:
+        """Call when a state is added to a domain."""
         self._cancel_listener(_ENTITIES_LISTENER)
         self._setup_entities_listener(
             self._last_track_states.domains, self._last_track_states.entities
         )
         self.hass.async_run_hass_job(self._action_as_hassjob, event)
 
     @callback
-    def _setup_domains_listener(self, domains: set[str]) -> None:
+    def _setup_entity_added_domain_listener(self, domains: set[str]) -> None:
```

Comment:
I renamed this since it wasn't clear that it only listens for new states in a specific domain to appear in the state machine which is why it likely led to the design that this PR adjusts